frame/scored-pool/src/tests.rs

```diff
@@ -297,3 +297,23 @@ fn candidacy_resubmitting_works() {
 		assert_eq!(ScoredPool::candidate_exists(who), true);
 	});
 }
+
+#[test]
+fn pool_candidates_exceded() {
+	new_test_ext().execute_with(|| {
+		for i in [1, 2, 3, 4, 6] {
+			let who = i as u64;
+			assert_ok!(ScoredPool::submit_candidacy(Origin::signed(who)));
+			let index = find_in_pool(who).expect("entity must be in pool") as u32;
+			assert_ok!(ScoredPool::score(Origin::signed(ScoreOrigin::get()), who, index, 99));
+		}
+
+		let submit_candidacy_call =
+			mock::Call::ScoredPool(crate::Call::<Test>::submit_candidacy {});
+
+		assert_noop!(
+			submit_candidacy_call.dispatch(Origin::signed(8)),
```

Comment:
https://github.com/paritytech/substrate/pull/11927 should make this superfluous; you can directly call the function without wrapping it.